client/jetpack-cloud/sections/pricing/jpcom-masterbar/components/mobile-menu-button.tsx

```diff
@@ -0,0 +1,68 @@
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useCallback } from 'react';
+import { closeOnFocusOut } from '../utils';
+import type { MouseEvent, FC, SetStateAction, Dispatch, RefObject } from 'react';
+
+interface MobileMenuButtonProps {
+	isOpen: boolean;
+	setIsOpen: Dispatch< SetStateAction< boolean > >;
+	mobileMenu: RefObject< HTMLDivElement >;
+}
+
+const MobileMenuButton: FC< MobileMenuButtonProps > = ( { isOpen, setIsOpen, mobileMenu } ) => {
+	const translate = useTranslate();
+
+	const mobileMenuToggle = useCallback( () => {
+		setIsOpen( ( open ) => ! open );
+
+		if ( ! isOpen ) {
+			document.body.classList.add( 'no-scroll' );
+		} else {
+			document.body.classList.remove( 'no-scroll' );
+		}
```

Comment:
```suggestion
		setIsOpen( ( open ) => {
		    		if ( open ) {
			            document.body.classList.add( 'no-scroll' );
		            } else {
			            document.body.classList.remove( 'no-scroll' );
		            }
		            return ! open;
		} );
```

I would propose to change class lists in the dispatcher. Or create a `useEffect( () => ..., [ isOpen ] )` outside of `useCallback` hook. I don't think we can depend on `isOpen` being updated synchronically during runtime of the function.

Reply:
I did not think of that 🤔 . I haven't seen any issues with the current way it is set up, but I think your way is safer